main.py

```diff
@@ -242,62 +266,61 @@ def main(cl_arguments):
     else:
         # Look for eval checkpoints (available only if we're restoring from a run that already
         # finished), then look for training checkpoints.
-        eval_best = glob.glob(os.path.join(args.run_dir,
-                                           "model_state_eval_best.th"))
-        if len(eval_best) > 0:
-            load_model_state(
-                model,
-                eval_best[0],
-                args.cuda,
-                task_names_to_avoid_loading,
-                strict=strict)
+        best_path = get_best_checkpoint_path(args.run_dir)
+        if best_path:
+            load_model_state(model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict)
         else:
-            macro_best = glob.glob(os.path.join(args.run_dir,
-                                                "model_state_main_epoch_*.best_macro.th"))
-            if len(macro_best) > 0:
-                assert_for_log(len(macro_best) == 1,
-                               "Too many best checkpoints. Something is wrong.")
-                load_model_state(
-                    model,
-                    macro_best[0],
-                    args.cuda,
-                    task_names_to_avoid_loading,
-                    strict=strict)
-            else:
-                assert_for_log(
-                    args.allow_untrained_encoder_parameters,
-                    "No best checkpoint found to evaluate.")
-                log.warning("Evaluating untrained encoder parameters!")
+            assert_for_log(args.allow_untrained_encoder_parameters,
+                           "No best checkpoint found to evaluate.")
+            log.warning("Evaluating untrained encoder parameters!")
```

Comment:
As in when there's no pretraining? Since if we do pretraining on BERT then the get_best_checkpoint_path function should return a non-empty string and it wouldn't fall into this case

Reply:
@W4ngatang

Reply:
Yea that's right. If you're just fine-tuning BERT on the target tasks (no intermediate task training), you'll hit this warning. We'll then create a checkpoint that we can reload at the start of each target task training step.